### ISSUE TITLE
v0.25.0 RC Post key by sending only the date part `YYYY-MM-DD` failed

### ISSUE THREAD

**Issue by alallema** [meilisearch/MeiliSearch#2017]
**Describe the bug**

Post a new key by sending only the date part for `expiresAt` field failed.
According to the [API Keys](https://github.com/meilisearch/specifications/blob/develop/text/0085-api-keys.md#155-post-patch---keys---api-key-object-payload-definition) specification and the API response, the `expiresAt` value can be defined only with the date such as: `YYYY-MM-DD`.

| field       | type    | required |description                     |
|-------------|---------|----------|--------------------------------|
| expiresAt   | string  | Required | The expiration date and time as `ISO-8601` format. `null` equals to no expiration time. Sending only the date part e.g `2021-12-01` leads to having an `expiresAt` value set to `2021-12-01T00:00:00`. **Default**: `No Default` |

**To Reproduce**

*Post on `/keys` route:*
```json
{
    "description": "Indexing Products API key",
    "actions": [
        "documents.add"
    ],
    "indexes": ["products"],
    "expiresAt": "2022-01-01"
}
```
*Response:*
```json
{
	"message": "expiresAt field value `\"2022-01-01\"` is invalid. It should be in ISO-8601 format to represents a date or datetime in the future or specified as a null value. e.g. 'YYYY-MM-DD' or 'YYYY-MM-DDTHH:MM:SS'.",
	"code": "invalid_api_key_expires_at",
	"type": "invalid_request",
	"link": "https://docs.meilisearch.com/errors#invalid_api_key_expires_at"
}
```

**Expected behavior**

The `expiredAt` must be able to be defined by a date format or the API message must be modified.

**MeiliSearch version:** v0.25.0rc1

**Comment by Kerollmops** [comment]
Hey @alallema,

It is indeed an issue on our side, as we must accept ISO-8601 date format we should accept the YYYY-MM-DD format and it doesn't seem to be the case. However, it seems like the chrono library that we use accepts RFC3339 instead, you can see [the difference on this website](https://ijmacd.github.io/rfc3339-iso8601/).

I would bet that we always try to [parse the string as a chrono DateTime](https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html) even if there is no time given in the string therefore the chrono library tells us that it is too short and it is right. We should have tried to [parse the string as a NaiveDate](https://docs.rs/chrono/0.4.19/chrono/naive/struct.NaiveDate.html) when the first fully-qualified string was parsed and failed, this way we can accept the YYYY-MM-DD format and specify that the time is in fact midnight and the offset is +0 (UTC). I have done some [tests on the Rust playground and it works](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3bab0fbf31b928dbf75b7f32ce1d931e).

**Comment by ManyTheFish** [comment]
Fixed by #2026
